pkg/controller/agent/pod.go

```diff
@@ -148,6 +148,8 @@ func buildPodTemplate(params Params, fleetCerts *certificates.CertificatesSecret
 		if builder, err = amendBuilderForFleetMode(params, fleetCerts, fleetToken, builder, configHash); err != nil {
 			return corev1.PodTemplateSpec{}, err
 		}
+		// Point agent to static config file mounted from a secret to /etc/agent/elastic-agent.yml
+		builder = builder.WithArgs("-e", "-c", path.Join(ConfigMountPath, ConfigFileName))
```

Comment:
```diff
diff --git a/pkg/controller/agent/pod.go b/pkg/controller/agent/pod.go
index 13e53078d..9ee674e73 100644
--- a/pkg/controller/agent/pod.go
+++ b/pkg/controller/agent/pod.go
@@ -450,7 +450,7 @@ if [[ -f %[1]s ]]; then
     %[5]s
   fi
 fi
-/usr/bin/tini -- /usr/local/bin/docker-entrypoint -e
+/usr/bin/tini -- /usr/local/bin/docker-entrypoint -e "$@"
 `, caPath, ubiSharedCAPath, ubiUpdateCmd, debianSharedCAPath, debianUpdateCmd)
 }
```
We need to make sure the arguments added here are actually propagated to the command.

Thanks to @pkoutsovasilis for helping me debug this.